### ISSUE TITLE
BOINC Manager freezes when trying to connect to remote computer

### ISSUE THREAD

**Issue by gopherit** [BOINC/boinc#3341]
(I couldn't find a duplicate issue.)

**Describe the bug**
When using a local BOINC Manager GUI to connect to a remote host, if the remote host is not available (for whatever reason), the local BOINC Manager freezes and must be force closed.

**Steps To Reproduce**
1.  File --> Select computer... --> Enter a remote computer
2.  Manager becomes 99% unresponsive to input.

**Expected behavior**
An error message e.g. "Cannot connect to remote computer" would be preferable to just freezing.

**System Information**
 - OS:  Win10 Pro x64, all versions (e.g. 1803, 1809, etc.)
 - BOINC Version:  Tested with 7.14.2 x64

**Comment by AenBleidd** [comment]
I'll take a look at it.
I have a clue why this happens, I'll try to reproduce it

**Comment by adamradocz** [comment]
I had the same issue with earlier versions too.

**Comment by AenBleidd** [comment]
@gopherit, @adamradocz, could you provide any additional details?
I was not able to reproduce this on BOINC 7.14.2 x64 on Windows 10 1903.
For me while trying either to access non-existing ip or existing ip that doesn't run BOINC client, BOINC Manager is responsive and works smooth without any significant CPU usage

**Comment by gopherit** [comment]
Yes, in this case the target machine is having DNS and other network problems.  Where is the Event Log stored in Linux?  I cannot find "stdoutdae.txt" in /var/lib/boinc-client.  I can SSH into the target machine and will try to read the log.  I think in the past the Event Log said "Cannot resolve host SourceMachine"

* SourceMachine (Manager):  7.14.2 x64 in Windows 10 trying to connect to TargetMachine
* TargetMachine (Client only):  7.14.2 x64 Linux with DNS problems

The SourceMachine Manager will "freeze" and then say "Connecting to TargetMachine" then "Disconnected" then "Connecting to TargetMachine" then "Disconnected" over and over.  boincmgr.exe CPU usage is 0-1% but I have to End Process anyways.

**Comment by adamradocz** [comment]
I had the problem when I tried to connect with the hostname not the IP address.

**Comment by gopherit** [comment]
For me, whether I specify the hostname or IP of the target machine in the "Host name" field gives me the same result.  I think the root cause is that remote_hosts.cfg must include only hostnames which requires the target machine to do a DNS query.  If DNS fails (such as the source machine not registering its hostname in DNS), then the target machine will reject the source machine, and the source machine will freak out and go into the Connecting/Disconnected loop.

**Comment by AenBleidd** [comment]
Similar issue from user:

> My main rig is Windows 10 x64 running BOINC 7.14.2. The server I connect to is a Linux Ubuntu 18.04 x64 server running BOINC 7.9.3 (most recent repo version, haven't been bothered to update manually)
> 
> I use both the machine name and it's local IP (192.168 range). For some reason mDNS isn't reliable on my network so i typically use the IP when I can. I usually encounter the bug by accident when I try to use the hostname but mDNS decided it wanted to be broken today, so it doesn't route. But if I fat-finger the IP or the host is turned off the same thing happens.
> 
> If the host goes offline while I'm still connected to it, the lag does NOT happen. I can click through the GUI just fine and tell it to connect to a different client. It only happens when I'm currently connected to a client, and then try to connect to an offline one.
